lib/vmpooler/api/v1.rb

```diff
@@ -496,17 +505,32 @@ def validate_date_str(date_str)
         jdata = JSON.parse(request.body.read)
 
         jdata.each do |param, arg|
+          failure = false
+
           case param
             when 'lifetime'
               arg = arg.to_i
 
               if arg > 0
                 $redis.hset('vmpooler__vm__' + params[:hostname], param, arg)
-
-                status 200
-                result['ok'] = true
+              else
+                failure = true
+              end
+            when 'tags'
+              if arg.is_a?(Hash)
+                arg.keys.each do |tag|
+                  $redis.hset('vmpooler__vm__' + params[:hostname], 'tag:' + tag, arg[tag])
+                end
+              else
+                failure = true
               end
           end
+
+          if failure
+            status 400
+          else
+            status 200
+            result['ok'] = true
         end
```

Comment:
Missing an `end`. Also, this `if` is within the `.each` block. This needs to be after that block, otherwise a successful param will overwrite `failure` to `false` at the top.